examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```diff
@@ -0,0 +1,180 @@
+   * of the streaming job is set to 40 seconds
+   * @param host Hostname that Spark Streaming context has to connect for receiving data
+   * @param port Port that Spark Streaming context has to connect for receiving data
+   * @param outputPath Output path to save the processed data read by the Spark Streaming context
+   */
+  def processStreamingWordCount(
+      host: String,
+      port: Int,
+      outputPath: String): ScalaJobHandle[Unit] = {
+    scalaClient.submit { context =>
+      context.createStreamingContext(20000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyContent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(35000)
+      ssc.stop(false, true)
+    }
+  }
+
+  /**
+   * Submits a spark sql job to the livy server
+   * <p/>
+   * The sql context job reads data frames from the given json path and executes
+   * a sql query to get the word with max count on the temp table created with data frames
+   * @param inputPath Input path to the json data containing the words
+   */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[String] = {
+    scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      try {
+        val rdd = sqlctx.read.json(inputPath)
+        rdd.registerTempTable("words")
+        val result = sqlctx.sql("select word, count(word) as word_count from words " +
+          "group by word order by word_count desc limit 1")
+        result.first().toString()
+      } catch {
+        case exception: IOException => {
+          throw new Exception("The input path does not have any data frame")
+        }
+      }
+    }
+  }
+
+  private def filterEmptyContent(text: String): Boolean = {
+    text != null && !text.isEmpty
+  }
+
+  private def tokenize(text : String) : Array[String] = {
+    text.toLowerCase.replaceAll("[^a-zA-Z0-9\\s]", "").split("\\s+")
+  }
+
+  private def stopClient(shutdownContext: Boolean): Unit = {
+    scalaClient.stop(shutdownContext)
+  }
+
+  /**
+   * Main method of the WordCount App
+   * STEPS FOR EXECUTION
+   * 1) 'livy.file.local-dir-whitelist' config has to be set in livy.conf file for the jars to be
+   * added to the session
+   *
+   * 2) Libraries to be added to classpath - livy-api-version-SNAPSHOT.jar,
+   * livy-scala-api-version-SNAPSHOT.jar, livy-client-http-version-SNAPSHOT.jar
+   *
+   * @param args
+   * arg(0) - Livy server url
+   * arg(1) - Output path to save the text read from the stream
+   */
+  def main(args: Array[String]) {
+    if (args.length < 2) {
+      throw new IllegalArgumentException("Number of args is less")
```

Comment:
nit: "Incorrect arguments." or something along those lines. You should also be checking for exactly 2, since that's what you expect.

(In fact you could just omit this block since if you don't have exactly 2 arguments, L165 will throw an exception.)